Clean chain specs

JoshOrndorff opened the pull request:
This PR cleans up some tech debt that had accumulated in the various chain spec files. The primary change is that the `testnet_genesis` function now supports mapping arbitrarily many author ids at genesis.

In order to clean this up I modified parachain staking's genesis config to take the candidates and the nominations as separate fields, which also makes that pallet's mock `ExtBuilder` cleaner.

Finally, with the mappings cleaned up, I changed the default mappings to `Alice -> Alith` and `Bob -> Baltithar` for local specs and just `Alice -> Alith` for dev specs. This reduces the special role that the Gerald account previously played. We should consider also switching the sudo account to be Alith on these chains.

## Checklist

- [ ] Does it require a purge of the network?
- [ ] You bumped the runtime version if there are breaking changes in the **runtime** ?
- [ ] Does it require changes in documentation/tutorials ?

JoshOrndorff commented on pallets/parachain-staking/Cargo.toml:
```diff
@@ -6,6 +6,7 @@ edition = "2018"
 description = "parachain staking pallet for collator selection and reward distribution"
 
 [dependencies]
+log = "0.4"
```
Oops, didn't end up needing this. It should be removed
```suggestion
```

JoshOrndorff commented on pallets/parachain-staking/Cargo.toml:
```diff
@@ -24,6 +25,7 @@ sp-core = { git = "https://github.com/paritytech/substrate", branch = "polkadot-
 [features]
 default = ["std"]
 std = [
+	"log/std",
```
```suggestion
```

JoshOrndorff commented on runtime/moonbase/tests/common/mod.rs:
```diff
@@ -149,15 +149,17 @@ impl ExtBuilder {
-		let mut stakers: Vec<(AccountId, Option<AccountId>, Balance)> = Vec::new();
+		let mut candidates: Vec<(AccountId, Balance)> = Vec::new();
+		let mut nominations: Vec<(AccountId, AccountId, Balance)> = Vec::new();
 		for collator in self.collators {
-			stakers.push((collator.0, None, collator.1));
+			candidates.push((collator.0, collator.1));
 		}
 		for nominator in self.nominators {
-			stakers.push((nominator.0, Some(nominator.1), nominator.2));
+			nominations.push((nominator.0, nominator.1, nominator.2));
 		}
```
Actually, this can probably all go, right?

JoshOrndorff replied:
And be sure to remove it from other runtimes as well.

crystalin replied:
I haven't really looked at the test, why would it have to go ?

girazoki commented on node/service/src/chain_spec/moonshadow.rs:
```diff
@@ -58,12 +58,14 @@ pub fn development_chain_spec(mnemonic: Option<String>, num_accounts: Option<u32
 				vec![(
-					AccountId::from_str("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").unwrap(),
-					None,
+					AccountId::from_str("f24FF3a9CF04c71Dbc94D0b566f7A27B94566cac").unwrap(),
```
Actually we should be able to insert the account-generate vec directly here.

girazoki replied:
```suggestion
					accounts[0],
```

girazoki commented on node/service/src/chain_spec/moonshadow.rs:
```diff
@@ -95,12 +97,23 @@ pub fn get_chain_spec(para_id: ParaId) -> ChainSpec {
+				vec![
+					// Alice -> Alith
+					(
+						AccountId::from_str("f24FF3a9CF04c71Dbc94D0b566f7A27B94566cac").unwrap(),
```
```suggestion
						accounts[0],
```

JoshOrndorff commented:
To summarize the changes in Gerald's role:

Sudo
Changed from Gerald to Alith in all specs

Parachain Staking
Changed from Gerald to Alith in dev specs
Changed from Gerald to Alith and Baltithar in local specs

Author Mapping
Changed from Alice -> Gerald to Alice -> Alith in dev specs
Changed from Alice -> Gerald to Alice -> Alith and Bob -> Baltithar in local specs

Balances
All 10 dev accounts plus Gerald are still funded in the dev spec as they always were
Alith and Baltithar are funded in local specs and Gerald is no longer funded.
